client/my-sites/domains/domain-management/list/domain-item.jsx

```diff
@@ -3,31 +3,105 @@
  */
 import React, { PureComponent } from 'react';
 import PropTypes from 'prop-types';
+import Gridicon from 'components/gridicon';
+import { localize } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
-import { CompactCard } from '@automattic/components';
+import { Button, CompactCard } from '@automattic/components';
+import FormCheckbox from 'components/forms/form-checkbox';
+import DomainNotice from 'my-sites/domains/domain-management/components/domain-notice';
 
 class DomainItem extends PureComponent {
 	static propTypes = {
 		domain: PropTypes.object.isRequired,
+		showSite: PropTypes.bool,
+		showCheckbox: PropTypes.bool,
 		onClick: PropTypes.func.isRequired,
+		onAddEmailClick: PropTypes.func.isRequired,
+		onToggle: PropTypes.func,
+	};
+
+	static defaultProps = {
+		showSite: false,
+		showCheckbox: false,
+		onToggle: null,
 	};
 
 	handleClick = () => {
 		this.props.onClick( this.props.domain );
 	};
 
+	stopPropagation = ( event ) => {
+		event.stopPropagation();
+	};
+
+	onToggle = ( event ) => {
+		if ( this.props.onToggle ) {
+			this.props.onToggle( event.target.checked );
+		}
+	};
+
+	addEmailClick = ( event ) => {
+		const { domain, onAddEmailClick } = this.props;
+		event.stopPropagation();
+		onAddEmailClick( domain );
+	};
+
 	render() {
+		const { domain, showSite, showCheckbox, translate } = this.props;
+
+		const iconClassName = 'card__link-indicator';
+
 		return (
-			<CompactCard className={ 'domain-item' } onClick={ this.handleClick }>
+			<CompactCard className="domain-item" onClick={ this.handleClick }>
+				<Gridicon className={ iconClassName } icon="chevron-right" />
+				{ showCheckbox && (
+					<FormCheckbox
+						className="domain-item__checkbox"
+						onChange={ this.onToggle }
+						onClick={ this.stopPropagation }
+					/>
+				) }
 				<div className="domain-item__link">
-					<div className="domain-item__title">{ this.props.domain.domain }</div>
+					<div className="domain-item__status">
+						<div className="domain-item__title">{ domain.domain }</div>
+						<DomainNotice status="info" text={ translate( 'Activating domain' ) } />
```

Comment:
If those strings are placeholders let's not put them in `translate` for now since they'll be sent for translation